code/datums/status_effects/debuffs.dm

```diff
@@ -67,8 +67,17 @@
 	return ..()
 
 /datum/status_effect/incapacitating/sleeping/tick()
-	if(owner.getStaminaLoss())
-		owner.adjustStaminaLoss(-0.5) //reduce stamina loss by 0.5 per tick, 10 per 2 seconds
+	if(owner.health > owner.crit_threshold)
+		var/healing = 15
+		if((locate(/obj/structure/bed) in owner.loc))
+			healing += 10
+		else
+			if((locate(/obj/structure/table) in owner.loc))
+				healing += 5
+		owner.adjustBruteLoss((-healing/50), 0)
```

Comment:
Choosing an arbitrary integer just so you can divide it by 50. Yep, sounds like something hippie would do

Reply:
Replace
-15 with 0.3
-10 with 0.2
-5 with 0.1
Get rid of
-Division

Reply:
Also, isn't the 0 a boolean? Use the defines.

Reply:
makes sense did some stuff